cdap-ui/app/features/hydrator/controllers/create/toppanel-ctrl.js

```diff
@@ -46,14 +53,90 @@ class TopPanelController{
     this.state = {
       metadata: {
         name: this.ConfigStore.getName(),
-        desciption: this.ConfigStore.getDescription()
+        description: this.ConfigStore.getDescription()
       },
       artifact: this.ConfigStore.getArtifact()
     };
   }
+
+  openMetadata() {
+    this.metadataExpanded = true;
+  }
+  resetMetadata() {
+    this.setState();
+    this.metadataExpanded = false;
+  }
+  saveMetadata() {
+    this.ConfigActionsFactory.setMetadataInfo(this.state.metadata.name, this.state.metadata.description);
+    this.metadataExpanded = false;
+  }
+  onEnterOnMetadata(event) {
+    // Save when user hits ENTER key.
+    if (event.keyCode === 13) {
+      this.saveMetadata();
+      this.metadataExpanded = false;
+    } else if (event.keyCode === 27) {
+      // Reset if the user hits ESC key.
+      this.resetMetadata();
+    }
+  }
+
+  onExport() {
+    let config = angular.copy(this.ConfigStore.getState());
+    this.$bootstrapModal.open({
+      templateUrl: '/assets/features/hydrator/templates/create/popovers/viewconfig.html',
+      size: 'lg',
+      keyboard: true,
+      controller: ['$scope', 'config', function($scope, config) {
+        $scope.config = JSON.stringify(config);
+
+        $scope.export = function () {
+          // CanvasFactory
+          //   .exportPipeline(
+          //     MyAppDAGService.getConfigForBackend(),
+          //     MyAppDAGService.metadata.name,
+          //     MyAppDAGService.nodes,
+          //     MyAppDAGService.connections)
+          //   .then(
+          //     function success(result) {
+          //       $scope.exportFileName = result.name;
+          //       $scope.url = result.url;
+          //       $scope.$on('$destroy', function () {
+          //         URL.revokeObjectURL($scope.url);
+          //       });
+          //       // Clicking on the hidden download button. #hack.
+          //       $timeout(function() {
+          //         document.getElementById('pipeline-export-config-link').click();
+          //       });
+          //     }.bind(this),
+          //     function error() {
+          //       console.log('ERROR: Exporting ' + MyAppDAGService.metadata.name + ' failed.');
+          //     }
+          //   );
+        };
+      }],
+      resolve: {
+        config: function() {
+          return config;
+        }
+      }
+    });
+  }
+  onSaveDraft() {
+
+  }
+  onValidate() {
+    this.ConsoleActionsFactory.addMessage({
```

Comment:
Are we resetting the messages in the console when we want to show new set of messages?  I believe ```this.ConsoleActionsFactory.resetMessage()``` would be called before adding the messages?